src/item.cpp

```diff
@@ -1947,12 +1947,18 @@ std::string item::info( std::vector<iteminfo> &info, const iteminfo_query *parts
     } ) ) {
 
         info.emplace_back( "QUALITIES", "", _( "Contains items with qualities:" ) );
-
+        std::map<quality_id, int> most_quality;
         for( const auto &e : contents ) {
             for( const auto &q : e.type->qualities ) {
-                name_quality( q );
+                //most_quality.emplace(q);
+                if( most_quality.operator[]( q.first ) < q.second ) {
+                    most_quality.insert_or_assign( q.first, q.second );
```

Comment:
The `operator[]` has already created the entry (if it did not exist), so this will never insert and always assign.

Suggestion:
```C++
if( most_quality.count( q.first ) ) {
     most_quality[q.first] = q.second;
} else {
     most_quality[q.first] = std::max( most_quality[q.first], q.second );
}
```

(Keep in mind that the `operator[]` creates a default constructed item, which is 0 for `int` and there can be negative qualities, e.g. for butchering.)

Reply:
i did something slightly different than the suggestion that does not require an if statement, hopefully it is similarly acceptable